### PR TITLE
Object Oriented JS

### PR DESCRIPTION
**Pull request by unabyband**
### Object Oriented JS

[Demo](https://unabyband.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/unabyband/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by unabyband** [comment]
Ok. I have to do some edits of constants and variables here.

**Comment by unabyband** [comment]
I've made some self-checking, review please

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@unabyband please copy/paste requirements posted/linked by bot. Go through them, mark every single one processed, fix code accordingly.
Ask specific questions here or in Students' chat if anything is not clear.

Why that? At least three requirements are not met.

**Comment by unabyband** [comment]
I'm not completely sure if I made all of necessary requirements. 
But, please review again, hope this looks much better now.
Thanks a lot!

**Comment by OleksiyRudenko** [comment]
@unabyband good job. A bit overkill on `x` and `y` but otherwise naming became clear. BTW, requirements contain a notice on `x` and `y` specifically.

Working with requirements is a skill. Luckily is well trainable. I am willing to invest some time into this.

Any requirements you are not sure you've met?

Copy paste them as a comment in this thread and add level of your "sureness" about them on the scal 0-5 or 0-10.

**Comment by unabyband** [comment]
Well, I edited methods followed by "Separation of Concerns" principe, hope they look more simple and also have clear naming now.

The next requirement:
"There are core constants and derived (calculated) constants (e.g. const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;)."
applies for other part of script, located in the file of the engine script (if I don't mistake).

Also, I don't realize how to use dt-parameter in player.update method... 
It could be obvious, but I understood how to used delta-time for enemies speed only.
Maybe this one of my mistakes. 
I'm just try working, developing friends-app and making review edits at the same time, so I'm really sorry for that.
Please re-review again
Best regards

File: submissions/unabyband/OO-JS/app.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+// You should multiply any movement by the dt parameter
+// which will ensure the game runs at the same speed for
+// all computers.
+Enemy.prototype.update = function(dt) {
+    this.horizontalMove += this.speed * dt;
+    if (this.horizontalMove > enemyEndPosition) {
+        this.horizontalMove = enemyStartPosition;
+        this.speed = speedIncreaser + Math.floor(Math.random() * 222);
+    };
+    this.checkCollision(player);
```
Comment (OleksiyRudenko) [inline submissions/unabyband/OO-JS/app.js:40]: See item 4.iii [here](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger)

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@unabyband one important fix and we are done.

File: submissions/unabyband/OO-JS/app.js (hunk 1 above)
Comment (unabyband) [reply submissions/unabyband/OO-JS/app.js]: Done, thanks for your notice

File: submissions/unabyband/OO-JS/app.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        player.verticalPosition + enemyHeight > this.verticalPosition) 
+        { 
+        if (winsCounter > 0) {
+            winsCounter -= 1;
+        } else { 
+            winsCounter = 0;
+            speedIncreaser = 100;
+        }
+            player.horizontalPosition = playerStartHorizontal;
+            player.verticalPosition = playerStartVertical;
```
Comment (OleksiyRudenko) [inline submissions/unabyband/OO-JS/app.js:61]: Now we are back to the problem with referring the global variable.
The requirements also suggest an apporach on how to resolve the issue.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@unabyband looks like a step back.
"Removal" of `player` reference doesn't automatically make issue resolved.
Feel free asking questions if anything in the requirements is not clear.

File: submissions/unabyband/OO-JS/app.js (hunk 2 above)
Comment (unabyband) [reply submissions/unabyband/OO-JS/app.js:62]: Ok, I've added an argument for tracking player position into Enemy class for checking collisions, hope this is the resolve of the issue (at least the requirements suggest exactly the same, as I understand).
Please, take a look again, thank you.

**Review by OleksiyRudenko — APPROVED** [review]
@unabyband wel done!
The less any function or class depend on global variables, the better. This helps to encapsulate business logic and less rely on "outer" world.

**Comment by unabyband** [comment]
It definitely make sense, I agree. Thank you

**Pull request merged**